File: pipeline/kite9-pipeline-common/src/main/java/com/kite9/k9server/adl/holder/pipeline/ADLDom.java
package com.kite9.k9server.adl.holder.pipeline;

import java.net.URI;

import org.kite9.diagram.dom.elements.ADLDocument;

import com.kite9.k9server.adl.format.media.DiagramFormat;

/**
 * Second point in the pipeline, the DOM can be worked on by commands.
 * 
 * @author robmoffat
 *
 */
public interface ADLDom extends XMLDom<ADLDocument> {

	/** 
	 * Does the actual processing
	 */
	<X extends DiagramFormat> ADLOutput<X> process(URI forLocation, X format);
	

	void ensureCssEngine(ADLDocument doc);
	
}

File: pipeline/kite9-pipeline-common/src/main/kotlin/com/kite9/k9server/adl/holder/pipeline/ADLDom.kt
package com.kite9.k9server.adl.holder.pipeline

import com.kite9.k9server.adl.format.media.DiagramWriteFormat
import java.net.URI

/**
 * Second point in the pipeline, the DOM can be worked on by commands.
 *
 * @author robmoffat
 */
interface ADLDom : XMLDom {

    /**
     * Does the actual processing: transforms into SVG and renders it.
     */
    fun <X> process(forLocation: URI, format: DiagramWriteFormat<X>): ADLOutput<X>

}